syft/internal/packagemetadata/names.go

```diff
@@ -0,0 +1,13 @@
+package packagemetadata
+
+import (
+	"reflect"
+)
+
+func AllNames() []string {
```

Comment:
question: I missed this in the sync review, but where the "incorrect" names being handled?

Reply:
this PR doesn't go all the way with the `packagemetadata`, that's reserved for https://github.com/anchore/syft/issues/1844 . In short, this isn't used yet (except for a test), only AllTypes() is used and is consistent with the `sourcemetadata` package approach. The value add for adding this in this PR was to add the completion test around `packagemetadata.AllNames()` to detect when there is drift with the ast analysis and the generated code... a test that will be extended to include the JSON name mapping in #1844 (as done in the `sourcemetadata.AllNames()` completion test).